client/rack_contoller.go

```diff
@@ -0,0 +1,159 @@
+	params := map[string]interface{}{}
+	err := r.client(systemID).Get("power_parameters", url.Values{}, func(data []byte) error {
+		return json.Unmarshal(data, &params)
+	})
+
+	return params, err
+}
+
+// PowerOn rack controller.
+func (r *RackController) PowerOn(systemID string, params *entity.RackControllerPowerOnParams) (*entity.RackController, error) {
+	qsp, err := query.Values(params)
+	if err != nil {
+		return nil, err
+	}
+
+	rackController := new(entity.RackController)
+	err = r.client(systemID).Post("power_on", qsp, func(data []byte) error {
+		return json.Unmarshal(data, rackController)
+	})
+
+	return rackController, err
+}
+
+// PowerOff rack controller.
+func (r *RackController) PowerOff(systemID string, params *entity.RackControllerPowerOffParams) (*entity.RackController, error) {
+	qsp, err := query.Values(params)
+	if err != nil {
+		return nil, err
+	}
+
+	rackController := new(entity.RackController)
+	err = r.client(systemID).Post("power_off", qsp, func(data []byte) error {
+		return json.Unmarshal(data, rackController)
+	})
+
+	return rackController, err
+}
+
+// GetPowerState of the rack controller.
+func (r *RackController) GetPowerState(systemID string) (*entity.RackControllerPowerState, error) {
+	ps := new(entity.RackControllerPowerState)
+	err := r.client(systemID).Get("query_power_state", url.Values{}, func(data []byte) error {
+		return json.Unmarshal(data, ps)
+	})
+
+	return ps, err
+}
+
+// Abort rack controller current operation.
+func (r *RackController) Abort(systemID string, comment string) (*entity.RackController, error) {
+	qsp := make(url.Values)
+	if comment != "" {
+		qsp.Set("comment", comment)
+	}
+
+	rackController := new(entity.RackController)
+	err := r.client(systemID).Post("abort", qsp, func(data []byte) error {
+		return json.Unmarshal(data, rackController)
+	})
+
+	return rackController, err
+}
+
+// Details gets the details for a given rack controller.
+func (r *RackController) Details(systemID string) (*entity.RackControllerDetails, error) {
+	rackControllerDetails := new(entity.RackControllerDetails)
+	err := r.client(systemID).Get("details", url.Values{}, func(data []byte) error {
+		return bson.Unmarshal(data, rackControllerDetails)
+	})
+
+	return rackControllerDetails, err
+}
+
+// OverrideFailedTesting ignores failed tests of the rack controller.
+func (r *RackController) OverrideFailedTesting(systemID string, comment string) (*entity.RackController, error) {
+	qsp := make(url.Values)
+	if comment != "" {
+		qsp.Set("comment", comment)
+	}
+
+	rackController := new(entity.RackController)
+	err := r.client(systemID).Post("override_failed_testing", qsp, func(data []byte) error {
+		return json.Unmarshal(data, rackController)
+	})
+
+	return rackController, err
+}
+
+// Test the rack controller.
+func (r *RackController) Test(systemID string, params map[string]interface{}) (*entity.RackController, error) {
```

Comment:
I wonder if we can change `map[string]interface{}` to a more concrete type like `RackControllerTestParams`?